raster/src/main/scala/geotrellis/raster/CellFeatures.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import geotrellis.raster.rasterize.Rasterizer
+import geotrellis.vector.{Feature, Geometry, Point, Polygon, Extent}
+import geotrellis.util.MethodExtensions
+
+import spire.syntax.cfor._
+
+
+/** Type class to convert a raster into features of cell geometries */
+trait CellFeatures[R, D] {
+
+  /** Describe cell grid of the source raster */
+  def cellGrid(raster: R): GridExtent[Long]
```

Comment:
This very clearly is crying out to be a type class of its own. `RasterLike` or something like that. That could be a meaningful filter on the `Methods` as well. If something can't be described with a `GridExtent[*]` we don't want to be talking about cell features.

IMO we can deal with that later. This method can be removed without impacting subclasses that may spring up in the meantime.